scripts/xtensa-build-zephyr.py

```diff
@@ -535,19 +543,27 @@ def build_platforms():
 		# when XTENSA_TOOLS_ROOT environmental variable is set,
 		# use user installed Xtensa tools not Zephyr SDK
 		if "XTENSA_TOOLS_VERSION" in platform_dict and xtensa_tools_root_dir:
+				# os.environ["XTENSA_TOOLS_VERSION"] = "RI-2022.10-win32"
+				# os.environ["XTENSA_CORE"] = "ace10_LX7HiFi4_2022_10"
+
 			xtensa_tools_root_dir = pathlib.Path(xtensa_tools_root_dir)
 			if not xtensa_tools_root_dir.is_dir():
 				raise RuntimeError(f"Platform {platform} uses Xtensa toolchain."
 					"\nVariable XTENSA_TOOLS_VERSION points path that does not exist\n"
 					"or is not a directory")
 
 			# set variables expected by zephyr/cmake/toolchain/xcc/generic.cmake
-			os.environ["ZEPHYR_TOOLCHAIN_VARIANT"] = "xcc"
+			if not os.environ["ZEPHYR_TOOLCHAIN_VARIANT"]:
```

Comment:
This should fix the build failures in 
```suggestion
			if not os.environ.get("ZEPHYR_TOOLCHAIN_VARIANT"):
```
```
    if not os.environ["ZEPHYR_TOOLCHAIN_VARIANT"]:
  File "/usr/lib/python3.8/os.py", line 675, in __getitem__
    raise KeyError(key) from None
```

Reply:
Or, even better
```suggestion
os.environ["ZEPHYR_TOOLCHAIN_VARIANT"] =
    os.environ.get("ZEPHYR_TOOLCHAIN_VARIANT", platf_stuff["DEFAULT_TOOLCHAIN_VARIANT"])
```

Reply:
Thanks marc, I'll fix that but I'm currently struggling to replace environmental variables for execute_command(build_cmd, cwd=west_top)... I guess i need to pass there my custom env

Reply:
This is currently failing on Linux SOFCI so better fix it sooner than later if you want some results?

I don't understand, we've been setting `os.environ["ZEPHYR_TOOLCHAIN_VARIANT"] = "xcc"` in this script since forever and it has been working since forever, at least on Linux...

Do you mean on Windows? I don't know about the environment on Windows (except it's very different from Linux) but on Linux all forked child processes automatically inherit the entire environment. That's why environment variable "hacks" are constantly abused on Linux: because it's so convenient.

Reply:
Marc I'm trying to hook up to our Internal CI.
The fix you send me is great and I already applied it but it is not enough for me to push, since it would trigger a working build... using xt-xcc (what is not my goal). Take a look at what our QB is doing with environmental variables before xtensa-build-zephyr.py is called. I need to override those to trigger xt-clang toolchain selection. I'll do this a bit later today since I confused timezones and I'm unable to operate any longer ;-)